test/base/test_history.py

```diff
@@ -79,13 +86,13 @@ def check_history(self):
                 if storage_type is not None:
                     read_results_from_file(
                         self.problem, self.history_options,
-                        len(result.optimize_result.list) + 2
+                        n_starts=n_starts,
```

Comment:
As indicated by the docstring here, having n_starts+2 was very much intentional

Reply:
I guess silently reading the same file multiple times bc no id is present, is not desirable, unless explicitly requested, thus the {id} check.

Reply:
but the `{id}` was there before. This is supposed to check whether read_results_from_file can handle missing files (optimization stopped before file was written, for whatever reason). If we do not wan't this behavior we should add a check whether `read_results_from_file` fails when the file is not present.

Reply:
how did the {id} come in before? It's not in the original file name in https://github.com/ICB-DCM/pyPESTO/commit/6d3226d6ccf841ad4c8f5cb77ca59f0fe79cd514#diff-80b64c062a6e89e507e7518b03d2cf45dc9518a824f39bae232b565e8e5016afL52, does it come in in processing somewhere?

Reply:
I thought it just read in the same file filename.csv 3 times in the previous test implementation, therefore all files existed. when adding {id}, then there was a trace is Null error or something.

Reply:
will fix this in a separate PR

Reply:
thanks :) it confused me too then. so in conclusion I guess yes we want to be able to merge histories also when some ids are not present (maybe warn then).